lib/rom/sql/extensions/postgres/commands.rb

```diff
@@ -12,22 +12,26 @@ module Create
           def insert(tuples)
             tuples.map do |tuple|
               relation.dataset.returning(*relation.columns).insert(tuple)
-            end.flatten
+            end.flatten.map { |t| relation.output_schema[t] }
           end
 
           # Executes multi_insert statement and returns inserted tuples
           #
           # @api private
           def multi_insert(tuples)
-            relation.dataset.returning(*relation.columns).multi_insert(tuples)
+            relation.dataset.returning(*relation.columns).multi_insert(tuples).map { |t|
+              relation.output_schema[t]
+            }
```

Comment:
heh, yeah, hooks, I didn't become handy with them yet >.< I'll give it a try, but why do we need to call `with_input_tuples` in `finalize`, those are _output_ tuples, aren't they?